components/pkg-export-container/src/naming.rs

```diff
@@ -97,11 +97,12 @@ impl Naming {
     // we are containerizing.
 
     /// Return the image name, along with a (possibly empty) vector of
-    /// additional tags.
+    /// additional bare tags, and a vector containing "name:tag"
+    /// identifiers (or just "name" if there are no tags).
     pub fn image_identifiers(&self,
                              ident: &FullyQualifiedPackageIdent,
                              channel: &ChannelIdent)
-                             -> Result<(String, Vec<String>)> {
+                             -> Result<(String, Vec<String>, Vec<String>)> {
```

Comment:
Would it be worth having a type alias or maybe even a new type for the different types of tags?

Reply:
Maybe... writing this return type killed a little bit of my soul.

I think that with some additional future refactorings this would get a lot simpler, but it may be a good idea to contain the gnarliness until then.

Reply:
Addressed in https://github.com/habitat-sh/habitat/pull/7682/commits/f057ddb7a2c74ef41ca2717d995e051020b23400; much better :smile: